src/Simulation/QsharpCore/Canon/NoOp.qs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Canon {
```

Comment:
Is this the namespace to keep them under, or would we even want them in Core?

Reply:
That would be an additional breaking change (one we could manage with a deprecation stub), such that I'd like to defer that for now if possible. That said, I think it makes sense to keep that under Canon so that it's guarded with an explicit open statement; NoOp is really useful in some cases, but not generically enough in my mind to elevate to auto-open.

Reply:
Ok.